docs/pages/custom-builds/functions.mdx

```diff
@@ -0,0 +1,284 @@
+
+EAS Build functions are a great way to extend the functionality of EAS Build custom builds. You can use them to write reusable pieces of custom build config in a very powerful and convenient way. This guide provides a walkthrough.
+
+<Step label="1">
+
+## Create EAS Build function
+
+The easiest way to create an EAS Build function is to use the `create-eas-build-function` CLI tool. By running the following command from the same directory as your **eas.json** file, you can create a new custom TypeScript function:
+
+<Terminal cmd={['$ npx create-eas-build-function@latest ./.eas/build/myFunction']} />
+
+This creates a new module called `myFunction` in the **.eas/build** directory. The module will contain a pre-generated module configuration and the **src** directory with the **index.ts** file containing the default TypeScript function template.
+
+```ts .eas/build/myFunction/src/index.ts
+// This file was autogenerated by `create-eas-build-function` command.
+// Go to README.md to learn more about how to write your own custom build functions.
+
+import { BuildStepContext } from '@expo/steps';
+
+// interface FunctionInputs {
+//   // specify the type of the inputs value and whether they are required here
+//   // example: name: BuildStepInput<BuildStepInputValueTypeName.STRING, true>;
+// }
+
+// interface FunctionOutputs {
+//   // specify the function outputs and whether they are required here
+//   // example: name: BuildStepOutput<true>;
+// }
+
+async function myFunction(
+  ctx: BuildStepContext
+  // {
+  //   inputs,
+  //   outputs,
+  //   env,
+  // }: {
+  //   inputs: FunctionInputs;
+  //   outputs: FunctionOutputs;
+  //   env: BuildStepEnv;
+  // }
+): Promise<void> {
+  ctx.logger.info('Hello from my TypeScript function!');
+}
+
+export default myFunction;
+```
+
+</Step>
+
+<Step label="2">
+
+## Build the EAS Build function and connect it to the config file
+
+After generating the custom build function, the next thing you need to do is to build it and connect it to your EAS Build custom builds config YAML file.
+
+### Build the EAS Build function
+
+To build the custom function you need to run the following command in **.eas/build/myFunction** directory:
+
+<Terminal cmd={['$ npm run build']} />
+
+This command triggers the `build` script placed in the **package.json** file of your custom function module.
+
+```json package.json
+{
+  ...
+  "scripts": {
+    ...
+    "build": "ncc build ./src/index.ts -o build/ --minify --no-cache --no-source-map-register" /* @end */
+    ...
+  },
+  ...
+}
+```
+
+> **Info** The `build` script uses the [ncc](https://github.com/vercel/ncc) to compile your custom function module into a single file with all its dependencies. If you don't have the `ncc` installed globally on your machine, run `npm install -g @vercel/ncc` to install it.
+
+The `build` script generates the **build/index.js** file that is used by the EAS Build to run your custom function.
```

Comment:
```suggestion
The `build` script generates **build/index.js**. This file must be uploaded to EAS Build as a part of your project archive, so that the builder can run your function. Ensure that the file is not excluded by a **.gitignore** file or **.easignore** file.
```